src/main/java/com/aws/greengrass/mqttbridge/MQTTBridge.java

```diff
@@ -149,8 +162,8 @@ public void startup() {
         }
 
         try {
-            if (mqttClient == null) {
-                mqttClient = new MQTTClient(this.config, mqttClientKeyStore, this.executorService);
+            if (mqttClient == null || mqttClient.isStopped()) {
```

Comment:
why do we need to create it again if it is stopped? Can't we just start it back up?

Reply:
issue is that the changed config is used in the constructor, so we'd need to recreate.  could recreate everytime, but looking through git history there seemed to be an issue with doing that, https://github.com/aws-greengrass/aws-greengrass-mqtt-bridge/commit/24132cca9c28cbc110ee6148549e28d379106979.  maybe with the recent addition of unsubscribing to `MQTTClientKeyStore`(in https://github.com/aws-greengrass/aws-greengrass-mqtt-bridge/pull/56/commits/d38ed00f5c79569ce00b5ed424df968ea0bf7655) that would fix it

Reply:
i did try moving `mqttClientInternal` initialization in `MQTTClient` to its `start` method, but that made unit testing cumbersome because it uses a mock for  `mqttClientInternal`